### DISCUSSION TITLE
How to restore a deleted VM

### DISCUSSION THREAD

**Discussion by jonathanlaniado** [utmapp/UTM#4279] [UTM Help (macOS)]
I thought it would show up in the Trash (macOS), but it's not there 😕

**Comment by veergalrani** [comment]
Same but i feel as if There is a special UTM Trash. I don't really know.

**Comment by aZealot** [comment]
I also want to know how to restore a VM I accidentally deleted. It had some stuff on there I didn't want to get rid of and I'm sad that I can't seem to get it back.

**Reply by osy** [reply]
Unfortunately this is not possible. This is why when you delete a vm a warning popup tells you that data will be lost permanently.
